src/dosbox.cpp

```diff
@@ -855,7 +855,7 @@ void DOSBOX_Init() {
 	// have worked with when tuning their game audio, we therefore use this
 	// same value given it's the most likely to produce audio as intended by
 	// the authors.
-	pint = secprop->Add_int("pcrate", when_idle, 18939);
+	pint = secprop->Add_int("pcrate", when_idle, 12000);
```

Comment:
Are you sure about this? As 18,939 seems to be oddly specific, most likely derived from the interrupt timer frequency?

I _swear_ I saw this frequency being mentioned somewhere in relation to DOSBox a few days ago, claiming that most DOS games that use the PC speaker need this rate to render audio correctly. I can't find it, but I've found something else in [this thread](https://www.vogons.org/viewtopic.php?t=43588):

> The 8253 clock is 1.193MHz (13.125/11), so as a 6-bit DAC your sample rate is **18.643kHz**. You can increase the sample rate at the expense of dynamic range (which is a strict quality improvement for properly dithered audio) up to 1-bit 596kHz audio.

>[...] most programers had PWM figured out by 1987, so the obvious examples (ie. Space Racer, games with RealSound like Mean Streets, etc.) all use the same basic PWM method. PWM on the PC was best implemented using timer channels 0 and 2, with 0 (the system timer) firing an interrupt at the desired sample rate; each time it fired, you sent a PCM sample value to PIT 2 which was tied to the speaker output. When PIT 2 got the value, speaker immediately went high, then after the count expired, went low again. Because this is happening at the rate of PIT 0, the PIT 2 manipulation has the effect of manipulating the width of the pulses, hence Pulse Width Modulation (PWM).
>
> The other method was to directly bit-bang the speaker, effectively treating it like a 1-bit PCM device, which earlier titles like Dunzhin used. That method produces very loud audio, but is quite distorted so it's only useful for speech.

This post is quite interesting and also mentions the 18.6kHz sample rate:
http://www.reenigne.org/blog/8088-pc-speaker-mod-player-how-its-done/

Another thread with tonnes of info about implementing a PC speaker emulation:
https://www.vogons.org/viewtopic.php?t=46255

It seems like it's best to leave the sample rate alone, or maybe set it to 18.643kHz? (but as it's so specific, probably the guy who wrote it knew what he was doing?)

Reply:
https://github.com/dosbox-staging/dosbox-staging/commit/7ede15f5706d1803c8ea84c9968374b4f50f95b5

I originally switched to this 19 KHz from 44.1 KHz.  which can help drop some of the higher frequency buzzing and harmonics.  The actual PIT rate is unaffected though, so this is more just about running the channel at a lower rate. 
Ripa's patch also has some harmonics and buzzing, which probably the original physical PC speaker "ate" by simply being a lower quality speaker mounted inside of a PC case. I found running it at 12000 Hz or 16000 Hz was about the right balance between chopping off the annoying frequencies while retaining enough signal. 
(But the user can still set pcrate however they like). 
Ultimately though, we will be able to do a lot better when your lowpass filters are in place!

Reply:
> The actual PIT rate is unaffected though, so this is more just about running the channel at a lower rate.

Right, I wasn't sure about this (didn't try to understand the whole code during the review).

> I found running it at 12000 Hz or 16000 Hz was about the right balance between chopping off the annoying frequencies while retaining enough signal.
> (But the user can still set pcrate however they like).

Seems like a sensible default then!

> Ultimately though, we will be able to do a lot better when your lowpass filters are in place!

I think so too; I found the small-speaker types need a lot of filtering to make them less annoying sounding (usually not higher than about 6k).